pkg/cvo/internal/operatorstatus.go

```
@@ -141,85 +141,66 @@ func waitForOperatorStatusToBeDone(ctx context.Context, interval time.Duration,
 			return false, nil
 		}
 
-		// undone is map of operand to tuple of (expected version, actual version)
-		// for incomplete operands.
-		undone := map[string][]string{}
+		if len(expected.Status.Versions) == 0 {
+			lastErr = &payload.UpdateError{
+				UpdateEffect: payload.UpdateEffectNone,
+				Reason:       "ClusterOperatorNotAvailable",
+				Message:      fmt.Sprintf("Cluster operator %s does not declare expected versions", expected.Name),
+				Name:         expected.Name,
+			}
+			return false, nil
```

Why do we not still want to check the operator's conditions? If the operator does not declare expected versions does that also imply that it also does not set conditions?

---

If the operator does not declare expected versions in its manifest, it is hugely bugged.  This PR is dropping the `Progressing` fallback to these cases, so I went with failing fast here to make sure we never let those bugs through CI.

---

@jottofar pointed out that for fail-fast, `UpdateEffectNone` should be `UpdateEffectFail`.  I've fixed that and rebased onto master with 4abd57c5 -> 98794a62.